supporting-docs/full-and-light-client-code-overlap.md

```diff
@@ -0,0 +1,49 @@
+Last edited: 18/07/19
+
+# Lodestar Full Client <> Light Client Architecture
+
+## Abstract
+This document outlines the structural differences between the Lodestar Full Client and Light Client. Clear distinctions will be made on the overlap between the two clients and how they share related packages. A description of each package is provided, in plain english (non-technical) where possible.
+
+
+## Directory Structure
+The following is the current directory structure for the Lodestar Full Client. Beside every directory is a hyphen with the an acknoweldgment if the Light Client will utilse it. The client is written in such a way that every directory is considered to be its own exporortable module, for those that know Golang, it mimics the package directory structure. This was planned months in advance in order to ensure that code can easily be shared between Lodestar tools and future devtooling tools alike. Modularity is key.
+```
+packages/lodestar/src
+├── chain - no
+├── cli - probably
+├── config - needed
+├── constants - needed
+├── db - needed
+├── eth1 - needed
+├── logger - needed
+├── network - potentially
+├── node - no
+├── opPool - no
+├── params - needed
+├── rpc - needed
+├── sszTypes - needed 
+├── sync - potnetially
+├── types - needed 
+├── util - potentially
+└── validator - no
+```
+
+## Module Definitions
+- **Chain:** The chain module contains the ETH2.0 consensus rules. This isn't used by the Light Client, but because the spec is frozen, and we are already at the latest version v0.8 there is very little to change here.
```

Comment:
```suggestion
- **Chain:** The chain module contains the ETH2.0 consensus rules. This isn't used by the Light Client, but because the spec is frozen, and we are already at the latest version v0.8, there is very little to change here.
```